src/components/TicTacToe/TicTacToe.mjs

```diff
@@ -0,0 +1,113 @@
+import css from "./TicTacToe.scss";
+import html from "./TicTacToe.html?raw";
+
+export class TicTacToe extends HTMLElement {
+    constructor() {
+        super();
+
+        // Create a shadow DOM
+        const shadow = this.attachShadow({ mode: "open" });
+        shadow.innerHTML = html;
+
+        const style = document.createElement("style");
+        style.textContent = css;
+        shadow.appendChild(style);
+
+        this.xWinsText = shadow.getElementById("x-win-local");
+        this.oWinsText = shadow.getElementById("o-win-local");
+
+        let storedResults = JSON.parse(localStorage.getItem("results"));
+
+        // Init default scores if it's the first time
+        if (!storedResults) {
+            storedResults = {
+                X: 0,
+                O: 0,
+            };
+            localStorage.setItem("results", JSON.stringify(storedResults));
+        }
+
+        this.updateLocalTextResults(storedResults);
+
+        const resetButton = shadow.getElementById("reset");
+        this.result = shadow.getElementById("result");
+        this.buttons = shadow.querySelectorAll(".game-button");
+        this.turn = 0;
+        this.remainingTurn = 9;
+        this.winningConditions = [
+            [0, 1, 2],
+            [3, 4, 5],
+            [6, 7, 8],
+            [0, 3, 6],
+            [1, 4, 7],
+            [2, 5, 8],
+            [0, 4, 8],
+            [2, 4, 6],
+        ];
+
+        this.result.textContent = "X player turn";
+        this.buttons.forEach((button) => {
```

Comment:
Priviligie l'usage des for..in et for..of, ils ont généralement de meilleures performances que les .forEach()
```js
for (button of this.buttons) {
  button.addEventListener("click", this.newTurn);
}
```